src/datasources/transaction-api/transaction-api.service.ts

```diff
@@ -831,6 +831,7 @@ export class TransactionApi implements ITransactionApi {
             offset: args.offset,
           },
         },
+        this.defaultExpirationTimeInSeconds,
```

Comment:
Context: sending a value here makes the underlying Redis command `EXPIRE` actually set a TTL for the cache registry. We were sending `undefined` to the Redis driver, which made the driver to immediately delete the registry, (from Redis docs):
```
Note that calling EXPIRE/[PEXPIRE](https://redis.io/commands/pexpire) with a non-positive timeout or [EXPIREAT](https://redis.io/commands/expireat)/[PEXPIREAT](https://redis.io/commands/pexpireat) with a time in the past will result in the key being [deleted](https://redis.io/commands/del) rather than expired
```

So this is the main change to store the `messages` in the cache.

Reply:
Unrelated but I think we'd benefit by having an anonymous interface for the arguments as with the number of `undefined` involved this can cause bugs in the future.

Reply:
💯 I'd add an anonymous interface to `CacheFirstDataSource` in a separate PR if you agree, since a lot of file would be affected 🙂